packages/webpack-cli/lib/groups/resolveConfig.js

```diff
@@ -204,7 +208,7 @@ const finalize = async (moduleObj, args) => {
         process.exit(2);
     }
 
-    newOptionsObject['options'] = isMultiCompilerMode ? configs : configs[0];
+    newOptionsObject['options'] = isMultiCompilerMode || isFunctionalConfig ? configs : configs[0];
```

Comment:
What if `Function` returns the one/multiple config(s)?

Reply:
Can you give an example?

Reply:
```js
module.exports = () => {
  return { entry }
}
```

and

```js
module.exports = () => {
  return [{ entry }, { entry }]
}
```

Reply:
I will update PR with tests for both cases.